frontend/src/pages/projects/notebook/useRouteForNotebook.ts

```diff
@@ -27,7 +27,7 @@ const useRouteForNotebook = (
       if (notebookName && projectName) {
         // if not using service mesh fetch openshift route, otherwise get Istio Ingress Gateway route
         const getRoutePromise = !enableServiceMesh
-          ? getRoute(notebookName, projectName).then((route) => route?.spec.host)
+          ? getRoute(notebookName, projectName).then((route) => route.spec.host)
           : getServiceMeshGwHost(projectName);
```

Comment:
If `enableServiceMesh` is `true`, but `getServiceMeshGwHost` fails to return a route, should it fallback to `getRoute` ?

Reply:
I don't think so. Since if a notebook has been created or restarted with `enableServiceMesh=true`, the notebook should not have oauth-proxy injected into it, so even if a route was successfully fetched with the `getRoute` call, I don't think that it would be functional. 

Potentially a user could start a notebook, leave it running and then install service mesh + enable the feature flag, and then be in a state where this scenario could apply. But from previous conversations, I don't think the goal is to support this partial Oauth/Istio state, the user should just restart the notebook in this case.